core/src/main/java/com/adobe/aem/sample/site/core/models/ComponentReportModel.java

```diff
@@ -24,26 +24,57 @@
 import java.util.List;
 import java.util.Map;
 
+/**
+ * The  ComponentReportModel class is a Sling Model used to get all the  components
+ * present in the specified AEM path. It retrieves component details and prepares a list of component
+ * reports.
+ */
 @Model(adaptables = {Resource.class, SlingHttpServletRequest.class}, defaultInjectionStrategy = DefaultInjectionStrategy.OPTIONAL)
 public class ComponentReportModel {
 
+    /**
+     * The title - title of Component
+     */
     @ValueMapValue
     private String title;
 
+    /**
+     * The dropDownPlaceHolder - Name of dropdown
+     */
     @ValueMapValue
     private String dropDownPlaceHolder;
 
+    /**
+     * The request -  SlingHttpServletRequest object
+     */
+    @SlingObject
+    private SlingHttpServletRequest request;
```

Comment:
do we need this object?

Reply:
Yess

Reply:
will your compoent  work in publish mode?

Reply:
sir when we are using SlingHttpServletRequest request then the component is not working on publish mode. But now i am taking resource resolver from ResourceresolverFactory  now it is working in publish mode